native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/http2/Http2ConnectionManager.java

```diff
@@ -110,15 +119,17 @@ private synchronized Http2ChannelPool.PerRouteConnectionPool createPerRouteConne
      * @param httpRoute the http route
      * @return Http2ClientChannel
      */
-    public Http2ClientChannel borrowChannel(HttpRoute httpRoute) {
+    public Http2ClientChannel fetchChannel(HttpRoute httpRoute) {
         Http2ChannelPool.PerRouteConnectionPool perRouteConnectionPool;
-        perRouteConnectionPool = getOrCreatePerRoutePool(this.http2ChannelPool, generateKey(httpRoute));
-
-        Http2ClientChannel http2ClientChannel = null;
-        if (perRouteConnectionPool != null) {
-            http2ClientChannel = perRouteConnectionPool.fetchTargetChannel();
+            synchronized (this) {
+            perRouteConnectionPool = getOrCreatePerRoutePool(this.http2ChannelPool, generateKey(httpRoute));
+
+            Http2ClientChannel http2ClientChannel = null;
+            if (perRouteConnectionPool != null) {
```

Comment:
We can check whether this is null and return early right?

Reply:
Updated